### PR TITLE
macOS support improvements

### PR DESCRIPTION
**Pull request by bkabrda**
List of things changed from #426:

* Updated README
* Made user to run agent under configurable
* Fixed plist to actually run agent under the configured user
* Refactor duplicated linux/macos code to a common file
* When running under user who is also currently signed in via GUI, ensure that the user LaunchAgent and login items are removed

@lc-applause I'd appreciate your review on this set of modifications (I don't think it needs to be super in-depth, we'll do that when reviewing for `main` branch; for now, I'm only interested to hear if the set of changes looks reasonable and if this still works for you).

### CODE REVIEW

**Review by alai97 — APPROVED** [review]
Looks good for docs!

**Comment by bkabrda** [comment]
Ping @lc-applause - any thoughts?

File: handlers/main-macos.yml [hunk 1]
```diff
@@ -4,6 +4,21 @@
 # However, this is here because this is a "handler-like" task that is dynamically
 # included by a handler task in handlers/main.yml.
 
+# NOTE: We don't use bootout/bootstrap here, because bootout can't wait for the operation
+# to finish and if it's in progress for a longer time, bootstrap fails. We use the old
+# unload/load combo because they actually wait.
```
Comment (ghost) [inline handlers/main-macos.yml:9]: good to know

File: tasks/agent-macos.yml [hunk 2]
```diff
@@ -1,179 +1,92 @@
...
-    mode: 0640
-    group: "{{ macos_user_data.gid }}"
-  become: yes
-  with_items: "{{ datadog_checks|list }}"
-  when: datadog_manage_config
-  notify: restart datadog-agent-macos
+- name: Remove user login item
+  shell:
+    cmd: |-
+      osascript -e 'tell application "System Events" to if login item "Datadog Agent" exists then delete login item "Datadog Agent"'
```
Comment (ghost) [inline tasks/agent-macos.yml:43]: 👍

**Review by ghost — APPROVED** [review]
I like the changes and I have no concerns - I ran this branch on one of our test machines and everything was fine. Thank you for seeing this to the end!

**Comment by bkabrda** [comment]
Thanks a lot @lc-applause! I'll merge this PR to the macos-support branch and open a PR for the main branch to get another review and then merge and release 🚀

**Pull request merged**